darshan-util/pydarshan/tests/test_plot_io_cost.py

```diff
@@ -0,0 +1,139 @@
+            "MPI-IO",
+            pd.DataFrame(
+                np.array([[0.0196372866, 0.134251, 0.0475]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            "STDIO",
+            pd.DataFrame(
+                np.array([[0.0, 0.0001022815, 0.0]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            "POSIX",
+            pd.DataFrame(
+                np.array([[0.0, 49.022266, 0.005518]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            "MPI-IO",
+            pd.DataFrame(
+                np.array([[0.0, 49.022411, 0.023936]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+    ],
+)
+def test_get_io_cost_df(report, mod, expected_df):
+    # regression test for `plot_io_cost.get_io_cost_df()`
+    actual_df = get_io_cost_df(report=report, mod_key=mod)
+    assert_frame_equal(actual_df, expected_df)
+
+
+@pytest.mark.parametrize("mod", ["POSIX", "MPI-IO", "STDIO"])
+@pytest.mark.parametrize(
+    "report, expected_ylims", [
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            [0.0, 1.0],
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            [0.0, 120.0],
+        ),
+    ],
+)
+def test_plot_io_cost_ylims(report, expected_ylims, mod):
+    # test the y limits for both axes for the IO cost stacked bar graph
+    fig = plot_io_cost(report=report, mod_key=mod)
+    for i, ax in enumerate(fig.axes):
+        # there are only 2 axes, the first being the "raw" data
+        # and the second being the normalized data (percent)
+        actual_ylims = ax.get_ylim()
+        if i == 0:
+            assert_array_equal(actual_ylims, expected_ylims)
+        else:
+            # normalized data is always the same
+            assert_array_equal(actual_ylims, [0.0, 100.0])
+
+@pytest.mark.parametrize("mod", ["POSIX", "MPI-IO", "STDIO"])
+@pytest.mark.parametrize(
+    "report, expected_yticks, expected_yticklabels", [
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            [0.0, 0.2, 0.4, 0.6, 0.8, 1.0],
+            ["0.0", "0.2", "0.4", "0.6", "0.8", "1.0"],
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            [0, 24, 48, 72, 96, 120],
+            ["0", "24", "48", "72", "96", "120"],
```

Comment:
Instead of maintaining these typed out lists of identical values can't you do something like this to generate the expected labels for any expected tick list within the test proper:

`expected_yticklabels = [str(i) for i in expected_yticks]`

Reply:
I added this simplification in the latest commit(s).